.github/workflows/build.yml

```diff
@@ -33,35 +45,56 @@ jobs:
       - name: Angular CLI install
         run: npm install -g --silent @angular/cli@13.3.10
 
-      - name: Npm install
+      - name: Set image tag & other env variables
+        run: |
+          VERSION_TAG=$(sed 's/.*"version": "\(.*\)".*/\1/;t;d' ./package.json)
+          echo "VERSION_TAG=$VERSION_TAG"
+
+          IMAGE_TAGS="${VERSION_TAG}.$(date '+%Y.%m.%d').${{ github.run_number }}"
+          echo "IMAGE_SEMANTIC_HASH=$IMAGE_TAGS"
+          echo "Tags: $IMAGE_TAGS"
+
+          echo "VERSION_TAG=$VERSION_TAG" >> $GITHUB_ENV
+          echo "IMAGE_SEMANTIC_HASH=$IMAGE_TAGS" >> $GITHUB_ENV
+          echo "IMAGE_TAGS=$IMAGE_TAGS" >> $GITHUB_ENV
+          echo "SHOULD_PUBLISH=$SHOULD_PUBLISH"
+
+      - name: Install Node Packages
         run: npm install
 
-      - name: Run tslint
-        continue-on-error: true
-        run: npx tslint src/**/*.ts
+      - name: Login to Azure registry
+        uses: docker/login-action@v3.0.0
+        if: ${{ env.SHOULD_PUBLISH }}
+        with:
+          registry: oaftech.azurecr.io
+          username: ${{ vars.ACR_PULL_USERNAME }}
+          password: ${{ secrets.ACR_PULL_PASSWORD }}
 
-      - name: Run build
-        continue-on-error: true
-        run: npm run build:prod
+      - name: Build and push an image to container registry
+        uses: docker/build-push-action@v5.1.0
+        if: ${{ env.SHOULD_PUBLISH == 'true'}}
```

Comment:
We should always build, since we want to validate that the iage build works.
However we should not always push.

Reply:
Fixed